src/lib.rs

```diff
@@ -15,15 +15,15 @@
 //! assert_eq!(decrypted_plaintext.unwrap(), plaintext);
 // ```
 
-use base64;
 use byteorder::ReadBytesExt;
-use getrandom;
-use openssl;
+use std::error::Error;
+use std::fmt::{self, Display};
 use std::io::{Cursor, Read};
 use std::time;
 
 const MAX_CLOCK_SKEW: u64 = 60;
 
+#[derive(Clone)]
```

Comment:
What's the use case for `Clone` on this?

Reply:
Generally, the use case is if you want to share an initialized `Fernet` across your application (perhaps even across thread boundaries). Currently, you have to either re-initialize the `Fernet` instance every time or wrap it in `Arc`/`Rc`.

My specific use case is storing a `MultiFernet` in actix-web's worker state/settings. The server runs multiple workers which all get a copy of the server settings. Since `MultiFernet` is not clonable, it needs to be wrapped in `Arc`:

https://github.com/mozilla-services/autopush-rs/blob/b08fdbdd61edbc817835ab0fb18f304fd6fda505/autoendpoint/src/server/mod.rs#L28

Reply:
Hmm, at the risk of emberassing myself, `&Fernet` should be `Sync`, right, so you should be able to share that between workers?

Reply:
Worker state needs to be `'static`. Either way though, `Clone` is a very common and useful thing to derive. I don't think there's any downsides to doing this?

Reply:
Hmm, so I can't decide if this is unreasonable or not, but as I'd written this, I'd had this general idea that we should avoid cloning keys, to minimize the number of places they existed in memory. Would `Box` + `&Fernet` address actix-web's requirement?

Reply:
Ok, I've been persuaded that most other crypto crates offer `Clone`, so no sense being the odd person out.